### PR TITLE
General Clean Up of Storage and Environments

### PR DESCRIPTION
**Pull request by joshmeek**
*Note*: This PR is being merged into the `env` branch and not `master`. The tests will not pass and the next PR into the `env` branch will be adding new tests and removing the old ones for environments.

## What does this PR change?
- Allow `flow_file_path` specification for Storage and Environments.
- General clean up of Storage and Environments.
- Docstrings for Storage and Environments.
- Removed all older environment information (classes, docs, serialization) BUT not tests.

## Why is this PR important?
One step closer to new environment/storage paradigm.

### CODE REVIEW

File: src/prefect/environments/storage/docker.py [hunk 1]
```diff
@@ -174,10 +245,12 @@ def create_dockerfile_object_from_base_image(
             print('Beginning health check...')
             from prefect.utilities.environments import from_file
 
-            local_env = from_file('/root/.prefect/flow_env.prefect')
+            local_env = from_file('{flow_file_path}')
```
Comment (cicdw) [inline src/prefect/environments/storage/docker.py]: this appears to be missing a `.format(flow_file_path=self.flow_file_path)`

File: src/prefect/environments/storage/docker.py [hunk 2]
```diff
@@ -232,10 +319,12 @@ def create_dockerfile_object_from_dockerfile(
...
```
Comment (cicdw) [inline src/prefect/environments/storage/docker.py]: same comment as above

File: src/prefect/environments/execution/base.py [hunk 3]
```diff
@@ -1,41 +1,54 @@
...
         pass
 
-    def process(self, storage: "prefect.environments.storage.Storage") -> None:
-        """"""
-        pass
-
-    def execute(self) -> None:
-        """
-        Executes the environment on any infrastructure created during setup
+    def setup(self, storage: "prefect.environments.storage.Storage") -> None:
```
Comment (cicdw) [inline src/prefect/environments/execution/base.py]: Oh and question hereL: when will `setup` require the storage to be provided? just curious.

File: src/prefect/environments/storage/docker.py (hunk 1 above)
Comment (joshmeek) [reply src/prefect/environments/storage/docker.py]: No it isn't :) look a couple lines below

File: src/prefect/environments/execution/base.py (hunk 3 above)
Comment (joshmeek) [reply src/prefect/environments/execution/base.py]: If I have a setup that wants to spin up preexisting dask workers using the flow's stored image. Is one example

File: src/prefect/environments/storage/docker.py (hunk 1 above)
Comment (cicdw) [reply src/prefect/environments/storage/docker.py]: Ahhhhh gotcha. 👍

**Comment by maikelpenz** [comment]
What is the replacement now for
`from prefect.utilities.environments import from_file` ?

**Comment by cicdw** [comment]
Hi @maikelpenz! If you were using `from_file` to save / load flows to / from local files, then I recommend using the new [LocalStorage](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/environments/storage/local.py#L13) class.  

Let me know if that does or does not fit your use case and I'd be happy to help more!

**Pull request merged**